ast/compile_test.go

```diff
@@ -1515,6 +1515,118 @@ func TestCompilerRewriteExprTerms(t *testing.T) {
 	}
 }
 
+func TestIllegalFunctionCallRewrite(t *testing.T) {
+	cases := []struct {
+		note           string
+		module         string
+		expectedErrors []string
+	}{
+		/*{
+					note: "function call override in function value",
+					module: `package test
+		foo(x) := x
+
+		p := foo(bar) {
+			#foo := 1
+			bar := 2
+		}`,
+					expectedErrors: []string{
+						"undefined function foo",
+					},
+				},*/
```

Comment:
This fix doesn't cover the more general cases outside of function calls in comprehension heads, e.g. the above case and:
```rego
package test

foo(x) = x

p {
	foo := 2
	foo(1) == 1
}
```
These cases will generate a `rego_type_error: undefined function` error instead.
We could extend the fix in this PR to also cover these cases, but to a greater impact to existing policies (they'd still fail, but with another error).

Reply:
OK to keep as-is I think.